src/mesh/RadioLibInterface.cpp

```diff
@@ -278,19 +278,28 @@ void RadioLibInterface::onNotify(uint32_t notification)
                     startReceive();      // try receiving this packet, afterwards we'll be trying to transmit again
                     setTransmitDelay();
                 } else {
-                    // Send any outgoing packets we have ready as fast as possible to keep the time between channel scan and
-                    // actual transmission as short as possible
-                    meshtastic_MeshPacket *txp = txQueue.dequeue();
+                    // check for any remaining delay, and abort if we're still supposed to be waiting
+                    meshtastic_MeshPacket *txp = txQueue.getFront();
                     assert(txp);
-                    bool sent = startSend(txp);
-                    if (sent) {
-                        // Packet has been sent, count it toward our TX airtime utilization.
-                        uint32_t xmitMsec = getPacketTime(txp);
-                        airTime->logAirtime(TX_LOG, xmitMsec);
+                    if (txp->tx_after > millis()) {
```

Comment:
Don't think this is still relevant if you agree with my other comment, however, I think we should already use `notifyLater()` before doing the channel scan. This is because the channel scan forces the radio to go out of receive mode, to standby and then to CAD mode. There is a brief window where you might miss receiving packets here do to the switching time. We should really only do this when we're about to transmit. 

Also, although it's probably negligible, we should really make the time after the channel scan and actual transmit as short as possible (see the comment on line 288), because during this time others might jump in and it results in a collision. So, we should try to keep the processing effort minimal here.

Reply:
Makes sense -  I'll adjust the flow accordingly.